tests/snuba/api/endpoints/test_organization_events_mep.py

```diff
@@ -2584,6 +2588,74 @@ def test_device_class_filter(self):
         assert meta["fields"]["device.class"] == "string"
 
 
+class OrganizationEventsMetricsEnhancedPerformanceEndpointTestWithOnDemandMetrics(
+    MetricsEnhancedPerformanceTestCase
+):
+    viewname = "sentry-api-0-organization-events"
+
+    def setUp(self):
+        super().setUp()
+
+    def do_request(self, query):
+        self.login_as(user=self.user)
+        url = reverse(
+            self.viewname,
+            kwargs={"organization_slug": self.organization.slug},
+        )
+        with self.feature(
+            {
+                "organizations:on-demand-metrics-extraction": True,  # XXX: We have coupling
+                "organizations:on-demand-metrics-extraction-widgets": True,
+            }
+        ):
+            return self.client.get(url, query, format="json")
+
+    def _test_is_metrics_extracted_data(
+        self, params: dict[str, Any], expected_on_demand_query: bool, dataset: str
+    ) -> None:
+        spec = OnDemandMetricSpec(
+            field="count()",
+            query="transaction.duration:>1s",
+            spec_type=MetricSpecType.DYNAMIC_QUERY,
+        )
+
+        self.store_on_demand_metric(1, spec=spec)
+        response = self.do_request(params)
+
+        assert response.status_code == 200, response.content
+        meta = response.data["meta"]
+        assert meta.get("isMetricsExtractedData", False) is expected_on_demand_query
+        assert meta["dataset"] == dataset
+
+        return meta
+
+    def test_is_metrics_extracted_data_is_included(self):
+        self._test_is_metrics_extracted_data(
+            {
+                "field": ["count()"],
+                "dataset": "metricsEnhanced",
+                "query": "transaction.duration:>=91",
+                "useOnDemandMetrics": "true",
+                "yAxis": "count()",
+            },
+            expected_on_demand_query=True,
+            dataset="metricsEnhanced",
+        )
+
+    def test_is_metrics_extracted_data_is_excluded(self):
+        self._test_is_metrics_extracted_data(
+            {
+                "field": ["count()"],
+                "query": "transaction.duration:>=91",
+                "useOnDemandMetrics": "true",
+                "yAxis": "count()",
+            },
+            # Since we do not include metricsEnhanced dataset it will fail
```

Comment:
@k-fish is this an acceptable way of forcing a fallback to discover? Or something else?

Ideally, I would prefer using `metricsEnhanced` since that is what the UI uses to call the BE.

Reply:
`metricsEnhanced` triggers the fallback code, otherwise the `discover` dataset is selected so you're not really testing that it fell back, you fell back yourself by sending `discover` (not sending `metricsEnhanced`). Pretty much the only way to fallback these days is to have a column that's not in the indexer. Maybe try a group-by that's randomly named and don't fire `store_metric` for that name, it could cause the indexer to throw an error.